docs/programming-guide.md

```diff
@@ -1099,6 +1099,9 @@ for details.
 </tr>
 </table>
 
+Spark RDD API also exposes asynchronous versions of some actions like foreach, collect, count etc. which return a handle immediately to the caller which can be used to wait for its completion.
```

Comment:
I have a few edits to suggest so let me just propose some text directly:

The Spark RDD API also exposes asynchronous versions of some actions, like `foreachAsync` for `foreach`, which immediately return a `FutureAction` to the caller instead of blocking on completion of the action. This can be used to manage or wait for the asynchronous execution of the action.